### PR TITLE
feat: Introduce notifyFetch verb

### LINKED ISSUE #931: Retrieve the AtNotification using the notification-id
**Is your feature request related to a problem? Please describe.**
- Feature request to provide capability to retrieve the AtNotification object using the notification-id. 

- At the moment, only way to retrieve the notifications is query using the `notify:list` which returns a list of notifications; but user cannot retrieve a specific notification using the notification id.

**Describe the solution you'd like**
- Provide a new variant of notify which enables to retrieve the AtNotification object using the notification id

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did**
- Implement notify fetch verb which returns AtNotification object.

**- How I did it**
 - Files Modified:
     - at_end2end_test: 
         1. Add test for fetching the at_notification using the notification id
         2. Updated the version number because json encode of error responses are not released to server
    - at_functional_test:
         1. Add functional test for fetching the notification when notification id is present
         2. Add functional test for fetching the notification when notification id is not present
   - at_server:
        1. notify_fetch_verb_handler: A new class to fetch the notification.
        2. verb_handler_manager: Create NotifyFetchVerbHandler instance on server start-up
        3. notify_list_verb_test: Add unit test for notify fetch verb handler
    - at_server_spec:
         1. Add NotifyFetch verb.

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by gkc** [comment]
Will review this afternoon or tomorrow

File: at_secondary/at_secondary_server/lib/src/verb/handler/notify_fetch_verb_handler.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  String _getJsonEncodedString(AtNotification atNotification) {
+    var atNotificationJson = atNotification.toJson();
+    for (MapEntry entry in atNotificationJson.entries) {
+      if (entry.value is! String) {
+        atNotificationJson[entry.key] = entry.value.toString();
+      }
+    }
+    return jsonEncode(atNotificationJson);
```
Comment (gkc) [inline at_secondary/at_secondary_server/lib/src/verb/handler/notify_fetch_verb_handler.dart:50]: Why all of the above? Looking at https://api.flutter.dev/flutter/dart-convert/jsonEncode.html, can't we just immediately `return jsonEncode(atNotificationJson);` ?

File: at_secondary/at_secondary_server/test/notify_list_verb_test.dart [hunk 2]
```diff
@@ -365,6 +365,49 @@ void main() {
     });
     tearDown(() async => await tearDownFunc());
   });
+
+  group('A group of test to verify notify fetch', () {
```
Comment (gkc) [inline at_secondary/at_secondary_server/test/notify_list_verb_test.dart:369]: Please add a test for non-existent notification ID

**Review by gkc — CHANGES_REQUESTED** [review]

File: at_secondary/at_secondary_server/lib/src/verb/handler/notify_fetch_verb_handler.dart (hunk 1 above)
Comment (sitaram-kalluri) [reply at_secondary/at_secondary_server/lib/src/verb/handler/notify_fetch_verb_handler.dart:50]: @gkc :  AtNotification contains fields that are not converted to JSON by default; For example fields of type DateTime and enum's. So, we need to extend logic to have toJson on all  enums and DateTime fields and then use `jsonEncode(atNotificationJson)`.

Will make the necessary changes required and keep you updated.

File: at_secondary/at_secondary_server/test/notify_list_verb_test.dart (hunk 2 above)
Comment (sitaram-kalluri) [reply at_secondary/at_secondary_server/test/notify_list_verb_test.dart:369]: Added test for non existent notification

**Review by gkc — APPROVED** [review]

**Pull request merged**